python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py

```diff
@@ -101,12 +106,23 @@ def _execUpdateAction(self):
                 else:
                     action.callback_Failure()
 
+            self._currently_running_command = False
+
 # Create the thread pool
 
 
 def _makeThreadPool(threadPoolSize=THREAD_POOL_SIZE, daemonic=True):
+    global ThreadPool
+    if ThreadPool and len(ThreadPool) != 0:
+        #from Ganga.Core.exceptions import GangaException
+        #raise GangaException("Cannot doubbly init the ThreadPool! ThreadPool already populated with threads")
+        log.debug("Found a thread pool already in existance, wiping it and startig again!")
+        for i in ThreadPool:
+            del i
+        del ThreadPool[:]
+        ThreadPool = []
```

Comment:
The threads should have been joined and exited and cleaned up within the Monitoring loop of the previous run. If they have not this is a definite bug in the shutdown methods.
The array is (or should be by design) empty but shouldn't be made larger by effectively restarting ganga. The array should contain 4 possible running threads if this is requested by the user not (4* number of times ganga started the monitoring).
Whether I delete it or simply not add to it should be immaterial but I'm wanting to add new threads as the threads from the prior run should have been terminated and destroyed not, simply restarted here.